miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -73,6 +73,27 @@
 
   private static final Logger log = LoggerFactory.getLogger(DefaultSampleService.class);
 
+  public static boolean isValidRelationship(Iterable<SampleValidRelationship> relations, Sample parent, Sample child) {
+    if (parent == null && !isDetailedSample(child)) {
+      return true; // Simple sample has no relationships.
+    }
+    if (!isDetailedSample(child) || !isDetailedSample(parent)) {
+      return false;
+    }
+    return isValidRelationship(
+        relations,
+        ((DetailedSample) parent).getSampleClass(),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "parent" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)

Reply:
Null parent is handled for plain samples. For detailed samples, it would make sense for this method to return true if `child` is an Identity and false otherwise